### PR TITLE
feat: support pinned upstream QEMU ACPI generation

### PR DESCRIPTION
**Pull request by jdrean**
## Summary
- add a `qemu:10.1.0` ACPI generation target for deployments built from upstream QEMU releases
- verify the upstream QEMU tarball against its pinned SHA-256 checksum
- keep the build reproducible on the pinned Ubuntu 26.04 base image
- allow Docker installations without buildx by avoiding the buildx-only `--progress` flag

## Motivation
Tinfoil production hosts run upstream QEMU 10.1.0 rather than an Ubuntu QEMU source package. The generic `boot_config.qemu` support can reproduce their ACPI topology, but the generator also needs to build the exact upstream QEMU release to produce byte-identical tables.

## Validation
- `cargo test --lib`
- `cargo build --release --manifest-path cli/Cargo.toml`
- regenerated a production-captured 8 vCPU / 16 GiB / 3-disk ACPI table byte-for-byte using `--create-acpi-tables qemu:10.1.0`

### CODE REVIEW

**Comment by sahason** [comment]
@jdrean I am reviewing this PR and will complete by tomorrow.

File: src/acpi.rs [hunk 1]
```diff
@@ -327,29 +327,44 @@ struct QemuPkg<'a> {
     version: &'a str,
     /// Immutable OCI digest of the base image (`sha256:...`).
     image_digest: &'static str,
+    base_image: &'static str,
+    source_checksum: &'static str,
```
Comment (sahason) [inline src/acpi.rs:331]: The fields  `base_image` and `source_checksum` exist in the struct with no documentation on when it is meaningful.

The `source_checksum`  is non-empty only for upstream should be stated.

File: src/acpi.rs [hunk 2]
```diff
@@ -327,29 +327,44 @@ struct QemuPkg<'a> {
...
+            "",
         ),
         "ubuntu:26.04" => (
             "main",
             "1:10.2.1+ds-1ubuntu4",
+            "ubuntu:26.04",
             "sha256:f3d28607ddd78734bb7f71f117f3c6706c666b8b76cbff7c9ff6e5718d46ff64",
+            "",
+        ),
+        "qemu:10.1.0" => (
```
Comment (sahason) [inline src/acpi.rs:351]: `qemu:10.1.0` is not a distribution. It is a version. This breaks the convention established by ubuntu:25.04 / ubuntu:26.04 (distro identifiers), conflates two orthogonal concepts, and means adding upstream 10.2.0 requires a new match arm plus a new hardcoded SHA-256. 

I suggest to use `--create-acpi-tables ubuntu:26.04 <upstream QEMU version>` like the supported option
`--create-acpi-tables ubuntu:26.04 1:10.2.1+ds-1ubuntu5`

Upstream QEMU version passed as an argument should match with the tagged upstream QEMU version format in upstream QEMU e.g v10.1.0.
See https://github.com/qemu/qemu/tags

File: src/acpi.rs [hunk 3]
```diff
@@ -327,29 +327,44 @@ struct QemuPkg<'a> {
...
+            "e0517349b50ca73ebec2fa85b06050d5c463ca65c738833bd8fc1f15f180be51",
         ),
         other => bail!(
-            "Unsupported distribution: {other}. Supported: ubuntu:25.04, ubuntu:26.04"
+            "Unsupported distribution: {other}. Supported: ubuntu:25.04, ubuntu:26.04, qemu:10.1.0"
         ),
     };
     Ok(QemuPkg {
         source,
         version: version_override.unwrap_or(default_version),
```
Comment (sahason) [inline src/acpi.rs:364]: `qemu_pkg_for("qemu:10.1.0", Some("10.2.0"))` produces a QemuPkg with version = "10.2.0" but source_checksum still holds the SHA-256 of 10.1.0. The Docker build will download qemu-10.2.0.tar.xz and immediately fail the sha256sum --check, with an error message that gives no hint about why. There is no guard in `qemu_pkg_for` that rejects version_override for upstream targets, and the new test `qemu_pkg_for_honors_version_override` doesn't cover this case at all.

File: README.md
Comment (sahason) [inline README.md:262]: Is it still required to install `docker-buildx-plugin` as the PR allow Docker installations without buildx by avoiding the buildx-only --progress flag?

File: src/acpi.rs
Comment (sahason) [inline src/acpi.rs:454]: Here should be an explicit "upstream" case

Comment (sahason) [inline src/acpi.rs:847]: No test coverage for qemu:10.1.0 here.

File: README.md [hunk 4]
```diff
@@ -85,6 +85,8 @@ Create `metadata.json` file with the below metadata:
 
 - `qemu` *(optional)*: Generic QEMU shape descriptor consumed by `--create-acpi-tables`. Each field is a thin pass-through to QEMU. When present, the QEMU command is built from this block plus a minimal set of measurement-related core flags (`-smp ... -m ... -bios ... -nodefaults -vga none -nographic -no-reboot`). When absent, the tool falls back to the Canonical direct-boot args. Fields:
   - `machine` *(required when block is present)*: value passed verbatim to `-machine`, e.g. `"q35,kernel_irqchip=split,memory-backend=mem0,smm=off,pic=off"`.
+  - `pci_hole64_start` *(optional)*: deterministic 64-bit PCI aperture start used only by the patched ACPI dumper. This reproduces the window that live VFIO BARs would establish without requiring those host devices during offline generation.
+  - `pci_hole64_end` *(optional)*: exclusive end of that aperture. Set this when VFIO BAR sizing expands the window beyond the configured `pci-hole64-size`.
```
Comment (sahason) [inline README.md:89]: Says nothing about the accepted format (decimal integer or 0x-prefixed hex or 0-prefixed octal?). An example will be good here.

Comment (sahason) [inline README.md:89]: Could you please add intergration test with non-null values for these in metadata.json?

File: README.md [hunk 5]
```diff
@@ -121,8 +123,14 @@ main archive:
 --create-acpi-tables ubuntu:26.04                              # default pin
 --create-acpi-tables ubuntu:26.04 1:10.2.1+ds-1ubuntu5         # explicit version
 --create-acpi-tables ubuntu:26.04 ""                           # current main-archive tip
+--create-acpi-tables qemu:10.1.0                              # pinned upstream release
```
Comment (sahason) [inline README.md:126]: Please see this comment https://github.com/virtee/tdx-measure/pull/70#discussion_r3851665520
